### PR TITLE
Switch to use harness-docker as a parent harness

### PR DESCRIPTION
**Pull request by andytson-inviqa**
While workspace only supports the harnessLayers atm and not auto configuration of them on `ws create`, it'd still be a big help to avoid all the duplication and upgrades needed to files harness-docker also manages.

One change though here is to use a multi-stage build instead of intermediate layers for building an nginx image for pipeline/production

### CODE REVIEW

File: application/overlay/_twig/Jenkinsfile/test.twig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+        stage('Test')  {
+            parallel {
+{% if not @('jenkins.tests.isolated') %}
+{% if @('services.lighthouse.enabled') %}
```
Comment (kierenevans) [inline application/overlay/_twig/Jenkinsfile/test.twig:4]: Does this need bool()?
```suggestion
{% if bool(@('services.lighthouse.enabled')) %}
```

Comment (andytson-inviqa) [reply application/overlay/_twig/Jenkinsfile/test.twig:4]: no, bool is a php harness thing as docker harness doesn't use yes's ever

Comment (andytson-inviqa) [reply application/overlay/_twig/Jenkinsfile/test.twig:4]: bool would be removed in the next major php harness version (whatever version number we call that, and prob the one where docker harness used), as was a deprecation

**Review by kierenevans — APPROVED** [review]

File: application/overlay/_twig/Jenkinsfile/test.twig (hunk 1 above)
Comment (andytson-inviqa) [reply application/overlay/_twig/Jenkinsfile/test.twig:4]: we could say though that bool() is a nice type check in any case to avoid WTFs

**Pull request merged**